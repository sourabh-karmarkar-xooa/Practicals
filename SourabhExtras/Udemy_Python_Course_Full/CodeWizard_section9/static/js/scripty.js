$(document).ready(function(){
    console.log("loaded")
    $.material.init();
    $(document).on("submit","#register-form", function(e){
        e.preventDefault();
        console.log("form submit")
    })
})

