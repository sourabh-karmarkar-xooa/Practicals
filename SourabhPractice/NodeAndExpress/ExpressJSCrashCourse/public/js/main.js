$(document).ready(function(){
    $('.deleteUser').on('click',deleteUser);
});

function deleteUser(){
     var confirmation=confirm('Are You Sure?');
     if(confirmation)
     {
         $.ajax({
             type:'DELETE',
             url:'/users/delete/'+$(this).data('id')
         }).done(function(){
            window.location.replace('/');
         });
         window.location.replace('/');
     }
     else
     {
         return false;
     }
}