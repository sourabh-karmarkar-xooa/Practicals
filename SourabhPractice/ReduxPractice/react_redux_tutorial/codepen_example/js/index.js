const {createStore} = Redux;

const initstate={
    todos:[],
    posts:[]
}

function myreducer(state=initstate,action){
    if(action.type=='ADD_TODO')
    {
        return {
            ...state,
            todos:[...state.todos,action.todo]
        }
    }
    if(action.type=='ADD_POST')
    {
        return {
            ...state,
            posts:[...state.posts,action.post]
        }
    }
    console.log(action);
    console.log(state);
}

const store=createStore(myreducer);
store.subscribe(()=>{
    console.log('state updated');
    console.log(store.getState());
});

store.dispatch({type:'ADD_TODO',todo:'Buy milk'});
store.dispatch({type:'ADD_TODO',todo:'Sleep Some More'});
store.dispatch({type:'ADD_POST',post:'Egg Hunt'});

