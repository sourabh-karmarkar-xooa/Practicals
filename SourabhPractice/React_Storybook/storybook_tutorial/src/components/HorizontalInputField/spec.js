import React from "react";
import renderer from "react-test-renderer";
import { HorizontalInputField } from "./index.js";

describe("<HorizontalInputField /> testing", () => {
  it("renders component", () => {
    const actual = renderer.create(
      <HorizontalInputField label="Test" name="test" />
    );
    expect(actual).toMatchSnapshot();
  });

  it("passes props through", () => {
    const actual = renderer.create(
      <HorizontalInputField
        id="input-field"
        label="Test 2:"
        name="email"
        type="email"
        onBlur={jest.fn()}
      />
    );
    expect(actual).toMatchSnapshot();
  });
});
