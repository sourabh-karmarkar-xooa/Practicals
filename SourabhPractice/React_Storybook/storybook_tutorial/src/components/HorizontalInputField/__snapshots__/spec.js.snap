// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HorizontalInputField /> testing passes props through 1`] = `
<div
  className="sc-htpNat iAbEgG"
>
  <label
    htmlFor="email"
  >
    Test 2:
  </label>
  <input
    className="form-control"
    id="input-field"
    name="email"
    onBlur={[MockFunction]}
    type="email"
  />
</div>
`;

exports[`<HorizontalInputField /> testing renders component 1`] = `
<div
  className="sc-htpNat iAbEgG"
>
  <label
    htmlFor="test"
  >
    Test
  </label>
  <input
    className="form-control"
    name="test"
  />
</div>
`;
